app/src/sharedTest/java/org/oppia/android/app/resumelesson/ResumeLessonActivityTest.kt

```diff
@@ -0,0 +1,232 @@
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = ResumeLessonActivityTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class ResumeLessonActivityTest {
+
+  private val internalProfileId: Int = 1
+  private val topicId: String = FRACTIONS_TOPIC_ID
+  private val storyId: String = FRACTIONS_STORY_ID_0
+  private val explorationId: String = FRACTIONS_EXPLORATION_ID_0
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var context: Context
+
+  @get:Rule
+  val resumeLessonActivityTestRule = ActivityTestRule(
+    ResumeLessonActivity::class.java,
+    /* initialTouchMode= */ true,
+    /* launchActivity= */ false
+  )
+
+  @get:Rule
+  val accessibilityTestRule = AccessibilityTestRule()
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+    Intents.release()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  fun testResumeLessonActivity_hasCorrectActivityLabel() {
+    resumeLessonActivityTestRule.launchActivity(createResumeLessonActivityIntent())
+    val title = resumeLessonActivityTestRule.activity.title
+
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    assertThat(title).isEqualTo(context.getString(R.string.resume_lesson_activity_title))
+  }
+
+  @Test
+  fun testResumeLessonActivity_toolbarTitleIsDisplayed() {
+    launch<ResumeLessonActivity>(createResumeLessonActivityIntent()).use {
+      onView(
+        allOf(
+          instanceOf(TextView::class.java),
+          withParent(withId(R.id.resume_lesson_activity_toolbar))
+        )
+      )
+        .check(matches(withText("Resume Lesson")))
+    }
+  }
+
+  @Test
+  fun testResumeLessonActivity_configurationChange_toolbarTitleIsDisplayed() {
+    launch<ResumeLessonActivity>(createResumeLessonActivityIntent()).use {
+      onView(isRoot()).perform(orientationLandscape())
+      onView(
```

Comment:
add `testCoroutineDispatchers.runCurrent()` above to stabilize the orientation change.

Reply:
done